src/oas3/transformers/servers.ts

```diff
@@ -7,13 +7,21 @@ export function translateToServers(servers: ServerObject[]): IServer[] {
   return servers.map(server => ({
     description: server.description,
     url: server.url,
-    variables: mapValues<Dictionary<ServerVariableObject>, INodeVariable>(
-      pickBy(server.variables, isServerVariableObject),
-      value => ({
-        default: String(value.default),
-        description: String(value.description),
-        enum: map(value.enum, String),
-      }),
-    ),
+    variables: translateServerVariables(server.variables),
   }));
 }
+
+export function translateServerVariables(variables: DeepPartial<{ [v: string]: ServerVariableObject }> | undefined) {
+  if (!variables) return {};
```

Comment:
The null check should be done by the caller, so let's make the parameter mandatory and let the caller figure it out

```suggestion
export function translateServerVariables(variables: DeepPartial<{ Dictionary<ServerVariableObject> }>) {
```

Reply:
Why is the type now a DeepPartial?

Reply:
I set it to DeepPartial because our document is a DeepPartial OpenApiObject, which I believe therefore makes the servers DeepPartials as well.

Reply:
Understood.

`DeepPartial` is a a modifier that relaxes a constraint, while in general we want to put as much constraints as possible — so that the TypeChecker can prevent mistakes. In such case, I would change the definition to be `variables: Dictionary<ServerVariableObject>`

Reply:
Ok you can ignore this — such thing is coming from the document. This is fine.